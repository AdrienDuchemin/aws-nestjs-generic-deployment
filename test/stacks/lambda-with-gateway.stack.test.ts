import { join } from 'path'
import { expect as expectCDK, SynthUtils, countResources } from '@aws-cdk/assert'
import { App } from '@aws-cdk/core'
import { LambdaWithGatewayStack, LambdaWithGatewayStackProps } from '../../src/stacks/lambda-with-gateway.stack'

describe('LambdaWithGatewayStack', () => {
  let stack: LambdaWithGatewayStack
  const props: LambdaWithGatewayStackProps = {
    lambdaCodePath: join(__dirname, './lambda'),
  }

  beforeAll(()=> {
    stack = new LambdaWithGatewayStack(new App(), 'LambdaWithGatewayStack', props)
  })

  it('should create one Lambda', () => {
    expectCDK(stack).to(countResources("AWS::Lambda::Function", 1))
  })

  it('should create one API GATEWAY', () => {
    expectCDK(stack).to(countResources("AWS::ApiGateway::RestApi", 1))
  })

  it('should not have changed', () => {
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
  })
})