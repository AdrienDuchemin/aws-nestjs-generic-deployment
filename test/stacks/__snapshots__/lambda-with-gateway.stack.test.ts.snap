// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LambdaWithGatewayStack should not have changed 1`] = `
Object {
  "Outputs": Object {
    "lambdaApiLambdaStackgatewayEndpointFE65ED0A": Object {
      "Value": Object {
        "Fn::Join": Array [
          "",
          Array [
            "https://",
            Object {
              "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
            },
            ".execute-api.",
            Object {
              "Ref": "AWS::Region",
            },
            ".",
            Object {
              "Ref": "AWS::URLSuffix",
            },
            "/",
            Object {
              "Ref": "lambdaApiLambdaStackgatewayDeploymentStageprodC163504D",
            },
            "/",
          ],
        ],
      },
    },
  },
  "Resources": Object {
    "lambda8B5974B5": Object {
      "DependsOn": Array [
        "lambdaServiceRole494E4CA6",
      ],
      "Properties": Object {
        "Code": Object {
          "ZipFile": "lambda",
        },
        "Handler": "handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "lambdaServiceRole494E4CA6",
            "Arn",
          ],
        },
        "Runtime": "nodejs12.x",
        "Timeout": 60,
      },
      "Type": "AWS::Lambda::Function",
    },
    "lambdaApiLambdaStackgateway0B09E5ED": Object {
      "Properties": Object {
        "Name": "ApiLambdaStack-gateway",
      },
      "Type": "AWS::ApiGateway::RestApi",
    },
    "lambdaApiLambdaStackgatewayANY720AA276": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "lambda8B5974B5",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Fn::GetAtt": Array [
            "lambdaApiLambdaStackgateway0B09E5ED",
            "RootResourceId",
          ],
        },
        "RestApiId": Object {
          "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "lambdaApiLambdaStackgatewayANYApiPermissionApiLambdaStacklambdaApiLambdaStackgateway75932681ANYDFEBEAB7": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "lambda8B5974B5",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
              },
              "/",
              Object {
                "Ref": "lambdaApiLambdaStackgatewayDeploymentStageprodC163504D",
              },
              "/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "lambdaApiLambdaStackgatewayANYApiPermissionTestApiLambdaStacklambdaApiLambdaStackgateway75932681ANY019AAEE5": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "lambda8B5974B5",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
              },
              "/test-invoke-stage/*/",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "lambdaApiLambdaStackgatewayAccount8A454B14": Object {
      "DependsOn": Array [
        "lambdaApiLambdaStackgateway0B09E5ED",
      ],
      "Properties": Object {
        "CloudWatchRoleArn": Object {
          "Fn::GetAtt": Array [
            "lambdaApiLambdaStackgatewayCloudWatchRole8541155D",
            "Arn",
          ],
        },
      },
      "Type": "AWS::ApiGateway::Account",
    },
    "lambdaApiLambdaStackgatewayCloudWatchRole8541155D": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "apigateway.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AmazonAPIGatewayPushToCloudWatchLogs",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
    "lambdaApiLambdaStackgatewayDeploymentE146A3FD15d53bb6c7211cb68e55ec3f87ce42c0": Object {
      "DependsOn": Array [
        "lambdaApiLambdaStackgatewayproxyANY83F14A45",
        "lambdaApiLambdaStackgatewayproxy17327CF6",
        "lambdaApiLambdaStackgatewayANY720AA276",
      ],
      "Properties": Object {
        "Description": "Automatically created by the RestApi construct",
        "RestApiId": Object {
          "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
        },
      },
      "Type": "AWS::ApiGateway::Deployment",
    },
    "lambdaApiLambdaStackgatewayDeploymentStageprodC163504D": Object {
      "Properties": Object {
        "DeploymentId": Object {
          "Ref": "lambdaApiLambdaStackgatewayDeploymentE146A3FD15d53bb6c7211cb68e55ec3f87ce42c0",
        },
        "RestApiId": Object {
          "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
        },
        "StageName": "prod",
      },
      "Type": "AWS::ApiGateway::Stage",
    },
    "lambdaApiLambdaStackgatewayproxy17327CF6": Object {
      "Properties": Object {
        "ParentId": Object {
          "Fn::GetAtt": Array [
            "lambdaApiLambdaStackgateway0B09E5ED",
            "RootResourceId",
          ],
        },
        "PathPart": "{proxy+}",
        "RestApiId": Object {
          "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
        },
      },
      "Type": "AWS::ApiGateway::Resource",
    },
    "lambdaApiLambdaStackgatewayproxyANY83F14A45": Object {
      "Properties": Object {
        "AuthorizationType": "NONE",
        "HttpMethod": "ANY",
        "Integration": Object {
          "IntegrationHttpMethod": "POST",
          "Type": "AWS_PROXY",
          "Uri": Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":apigateway:",
                Object {
                  "Ref": "AWS::Region",
                },
                ":lambda:path/2015-03-31/functions/",
                Object {
                  "Fn::GetAtt": Array [
                    "lambda8B5974B5",
                    "Arn",
                  ],
                },
                "/invocations",
              ],
            ],
          },
        },
        "ResourceId": Object {
          "Ref": "lambdaApiLambdaStackgatewayproxy17327CF6",
        },
        "RestApiId": Object {
          "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
        },
      },
      "Type": "AWS::ApiGateway::Method",
    },
    "lambdaApiLambdaStackgatewayproxyANYApiPermissionApiLambdaStacklambdaApiLambdaStackgateway75932681ANYproxy71FF299D": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "lambda8B5974B5",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
              },
              "/",
              Object {
                "Ref": "lambdaApiLambdaStackgatewayDeploymentStageprodC163504D",
              },
              "/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "lambdaApiLambdaStackgatewayproxyANYApiPermissionTestApiLambdaStacklambdaApiLambdaStackgateway75932681ANYproxy9A8C8BE8": Object {
      "Properties": Object {
        "Action": "lambda:InvokeFunction",
        "FunctionName": Object {
          "Fn::GetAtt": Array [
            "lambda8B5974B5",
            "Arn",
          ],
        },
        "Principal": "apigateway.amazonaws.com",
        "SourceArn": Object {
          "Fn::Join": Array [
            "",
            Array [
              "arn:",
              Object {
                "Ref": "AWS::Partition",
              },
              ":execute-api:",
              Object {
                "Ref": "AWS::Region",
              },
              ":",
              Object {
                "Ref": "AWS::AccountId",
              },
              ":",
              Object {
                "Ref": "lambdaApiLambdaStackgateway0B09E5ED",
              },
              "/test-invoke-stage/*/{proxy+}",
            ],
          ],
        },
      },
      "Type": "AWS::Lambda::Permission",
    },
    "lambdaServiceRole494E4CA6": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
